File: programmers/etc/level3/최고의_집합.py
def solution(n, s):
    if n > s: 
        return [-1]
    d, m = divmod(s, n)
    return [d]*(n-m)+[d+1]*m


n, s = 2, 9
print(solution(n, s)) # [4, 5]

n, s = 2, 1
print(solution(n, s)) # [-1]

n, s = 2, 8
print(solution(n, s)) # [4, 4]

File: Java/programmers/etc/level3/최고의집합.java
package CodeTest.Java.programmers.etc.level3;

public class 최고의집합 {
    public int[] solution(int n, int s) {
        if (n > s) return new int[]{-1};
        
        int d = s/n, m = s%n;
        
        int[] answer = new int[n];
        for (int i=0; i<n-m; i++) answer[i] = d;
        for (int i=n-m; i<n; i++) answer[i] = d+1;
        
        return answer;
    }
}
